### PR TITLE
Apply Rector suggestions for PHP 8.1

### PR DESCRIPTION
**Pull request by asumaran**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Subset of WPCOM changes for PHP 8.1 compatibility based on the D129122-code diff.

Ref: p7H4VZ-4x1-p2

### How to test the changes in this Pull Request:

The changes on this PR are based on changes currently made in WPCOM, As there are no logic changes, only compatibility with PHP 8.1, it was deemed unnecessary to include testing instructions. To put it simply, the current tests should be able to complete without any problems, and that should be sufficient.

Anyway, here's the testing instructions.

**plugins/woocommerce/includes/legacy/api/v1/class-wc-api-server.php**

- Enable the Legacy API in WooCommerce > Settings > Advanced > Legacy API.
- Visit `http://site.tld/wc-api/v1` should respond with JSON.

**plugins/woocommerce/includes/legacy/api/v2/class-wc-api-server.php**

- Enable the Legacy API in WooCommerce > Settings > Advanced > Legacy API.
- Visit `http://site.tld/wc-api/v2` should respond with JSON.

**plugins/woocommerce/includes/legacy/api/v3/class-wc-api-server.php**

- Enable the Legacy API in WooCommerce > Settings > Advanced > Legacy API.
- Visit `http://site.tld/wc-api/v3` should respond with JSON.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/changelog/43227-as-diff-D129122 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: dev
+
+Apply Rector suggestions for PHP 8.1
```
Comment (prettyboymp) [inline plugins/woocommerce/changelog/43227-as-diff-D129122]: For all of these diffs, we can probably leave out the reference to Rector, even though that was the tool used to identify the issues.  Something along the lines of "Improving return value type checks prior to usage."

**Review by prettyboymp — COMMENTED** [review]
Changes look good.

**Review by prettyboymp — APPROVED** [review]
Changes look good.

File: plugins/woocommerce/changelog/43227-as-diff-D129122 (hunk 1 above)
Comment (asumaran) [reply plugins/woocommerce/changelog/43227-as-diff-D129122]: done https://github.com/woocommerce/woocommerce/pull/43227/commits/b56bfa1e537f9b8eecd13592faf6b61ddbdab93e

**Review by vedanshujain — APPROVED** [review]

**Pull request merged**